app/scripts/lib/rpc-method-middleware/handlers/wallet-requestPermissions.test.ts

```diff
@@ -0,0 +1,756 @@
+import {
+  invalidParams,
+  RequestedPermissions,
+} from '@metamask/permission-controller';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+} from '@metamask/multichain';
+import * as Multichain from '@metamask/multichain';
+import { Json, JsonRpcRequest, PendingJsonRpcResponse } from '@metamask/utils';
+import {
+  CaveatTypes,
+  RestrictedMethods,
+} from '../../../../../shared/constants/permissions';
+import { PermissionNames } from '../../../controllers/permissions';
+import { requestPermissionsHandler } from './wallet-requestPermissions';
+
+jest.mock('@metamask/multichain', () => ({
+  ...jest.requireActual('@metamask/multichain'),
+  setEthAccounts: jest.fn(),
+  setPermittedEthChainIds: jest.fn(),
+}));
+const MockMultichain = jest.mocked(Multichain);
+
+const getBaseRequest = () => ({
+  jsonrpc: '2.0' as const,
+  id: 0,
+  method: 'wallet_requestPermissions',
+  networkClientId: 'mainnet',
+  origin: 'http://test.com',
+  params: [
+    {
+      eth_accounts: {},
+      [Caip25EndowmentPermissionName]: {},
+      otherPermission: {},
+    },
+  ],
+});
```

Comment:
Nit: Since this is a function what are your thoughts on having this taking an `overrides` argument that could get mixed into the default object here? Perhaps this function could then be called `getMockRequest` or something of that nature. That would turns constructions like `{ ...getBaseRequest(), origin: ... }` into `getMockRequest({ origin: ... })`.

``` suggestion
const getBaseRequest = (overrides = {}) => ({
  jsonrpc: '2.0' as const,
  id: 0,
  method: 'wallet_requestPermissions',
  networkClientId: 'mainnet',
  origin: 'http://test.com',
  params: [
    {
      eth_accounts: {},
      [Caip25EndowmentPermissionName]: {},
      otherPermission: {},
    },
  ],
  ...overrides,
});
```

Reply:
https://github.com/MetaMask/metamask-extension/pull/27847/commits/ed503e4cb43f06b9eae4391359e5e56901f2931b